pkg/telemetry/storage/db_test.go

```diff
@@ -10,19 +10,29 @@ import (
 
 	"github.com/google/uuid"
 	"github.com/stretchr/testify/require"
+	"github.com/textileio/go-tableland/internal/tableland"
 	"github.com/textileio/go-tableland/pkg/telemetry"
 	"github.com/textileio/go-tableland/pkg/telemetry/publisher"
 
 	"github.com/textileio/go-tableland/tests"
 )
 
 func TestCollectAndFetchAndPublish(t *testing.T) {
-	t.Run("state hash", func(t *testing.T) {
-		dbURI := tests.Sqlite3URI(t)
-		s, err := New(dbURI)
-		require.NoError(t, err)
-		telemetry.SetMetricStore(s)
+	t.Parallel()
+
+	// Notes:
+	// This can't be wired per sub-tests for two reasons:
+	// 1- `telemetry.SetMetricStore(...)` is a global setup at the package level, and
+	// 2- `SetMetricStore(...)` has a `sync.Once` wrapping so can't be called more than once, so each sub-test can't
+	//     override their value.
+	//
+	// This also means that sub-tests can't run in parallel.
```

Comment:
Before this PR, this test only had a single sub-test.
Adding extra sub-tests had some surprising effects, explained in this comment.

It is not a big deal, but we can't strictly spin up a "complete fresh stack" per sub-test for the explained reasons.

LMK if it makes sense.

Reply:
yeah. makes sense. that's the trade-off of having those global variables